src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,1041 @@
+            this.whetherIncrementShouldIncrement = true
+        }
+    }
+
+    fillGridCell() {
+        this.drawSquare(this.primaryProperties, this.scalingFactor)
+        this.drawSquare(
+            this.secondaryProperties,
+            this.scalingFactor / 2,
+            this.scalingFactor / 4
+        )
+        if (this.showNumbers) {
+            this.showNumber()
+        }
+    }
+
+    drawSquare(props: number[], size: number, offset = 0) {
+        if (this.colorProperties(props)) {
+            this.sketch.rect(this.x + offset, this.y + offset, size, size)
+        }
+    }
+
+    propertiesFilledWith(fillType: PropertyVisualization) {
+        const retVal: number[] = []
+        for (let i = 0; i < this.propertyObjects.length; i++) {
+            if (
+                this.propertyObjects[i].property != Property.None
+                && this.propertyObjects[i].visualization === fillType
+            ) {
+                retVal.push(i)
+            }
+        }
+        return retVal
+    }
+
+    // returns whether any of the properties held, i.e. whether indicator
+    // needs to be drawn, and by side effect sets the fill color
+    colorProperties(props: number[]): boolean {
+        let retval = false
+        for (const i of props) {
+            if (
+                this.hasProperty(
+                    this.currentIndex,
+                    this.propertyObjects[i].property,
+                    this.propertyObjects[i].aux
+                )
+            ) {
+                this.sketch.fill(this.propertyObjects[i].color)
+                retval = true
+            }
+        }
+        return retval
+    }
+
+    hasProperty(ind: number, property: Property, aux?: bigint) {
+        const propertyName = Property[property] as PropertyName
+        if (propertyName in propertyOfFactorization) {
+            let factors: Factorization = null
+            if (this.currentNumber === this.seq.getElement(ind)) {
+                factors = this.seq.getFactors(ind)
+            }
+            if (factors === null) {
+                factors = simpleFactor(this.currentNumber)
+            }
+            if (factors === null) {
+                throw new RangeError(
+                    `Factorization needed to compute ${propertyName} `
+                        + `of ${this.currentNumber}, but unavailable.`
+                )
+            }
+            return propertyIndicatorFunction[
+                propertyName as FactorPropertyName
+            ](factors)
+        }
+        return propertyIndicatorFunction[propertyName as ValuePropertyName](
+            this.currentNumber,
+            aux
+        )
+    }
+
+    showNumber() {
+        const currentNumberAsString = this.currentNumber.toString()
+
+        if (this.showNumbers) {
+            this.sketch.fill(this.numberColor)
+            this.sketch.text(
+                currentNumberAsString,
+                this.x + 0 * this.scalingFactor,
```

Comment:
Why are we multiplying the scaling factor by 0? Why are we adding 0 to `this.x`?